lib/common/services/commands-service.ts

```diff
@@ -154,7 +153,7 @@ export class CommandsService implements ICommandsService {
 				return true;
 			}
 
-			this.$errors.fail("Unable to execute command '%s'. Use '$ %s %s --help' for help.", beautifiedName, this.$staticConfig.CLIENT_NAME.toLowerCase(), beautifiedName);
+			this.$errors.failWithHelp("Unable to execute command '%s'. Use '$ %s %s --help' for help.", beautifiedName, this.$staticConfig.CLIENT_NAME.toLowerCase(), beautifiedName);
```

Comment:
The error messages says to execute `tns ... --help` , but it also prints the help. I'm also not sure how it is possible to get here.
And finally, you can rewrite these several lines in:
```TypeScript
return await this.$injector.isValidHierarchicalCommand(commandName, commandArguments) || 
    await this.validateCommandArguments(command, commandArguments) ||
    this.$errors.fail(`Unable to execute command '${beautifiedName}'.`);
```